src/main/java/com/geespring/marketapi/repository/UserRepository.java

```diff
@@ -0,0 +1,15 @@
+package com.geespring.marketapi.repository;
+
+import com.geespring.marketapi.model.User;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+
+@Repository
+public interface UserRepository extends CrudRepository<User, Long> {
+
+    @Query("SELECT u FROM User u WHERE u.email = ?1")
```

Comment:
Most of these manual queries should be unnecessary because Spring can figure them out by method name, parameters, and return type via black magic.